tests/common/test_os_calls.c

```diff
@@ -0,0 +1,51 @@
+
+#if defined(HAVE_CONFIG_H)
+#include "config_ac.h"
+#endif
+
+#include <stdlib.h>
+#include "test_common.h"
+#include "os_calls.h"
+
+START_TEST(test_g_file_get_size__returns_file_size)
+{
+    unsigned long long size;
+
+    size = g_file_get_size("../../xrdp/xrdp256.bmp");
+    ck_assert_int_eq(size, 49278);
+
+
+    size = g_file_get_size("../../xrdp/ad256.bmp");
+    ck_assert_int_eq(size, 19766);
+
+}
+END_TEST
+
+START_TEST(test_g_file_get_size__more_than_4GiB)
+{
+    unsigned long long size;
+
+    system("dd if=/dev/zero of=./file_5GiB.dat bs=4k count=1310720");
```

Comment:
I'm not sure this is the correct way to create files required to perform tests. Let me know if there's an appropriate method.

Reply:
On my system I'm getting:-

```
FAIL: test_common 18 - test_os_calls.c:oscalls-file:test_g_file_get_size__more_than_4GiB: Test timeout expired
```

but, then my test system is using a mirror of hard drives.

Also, you're leaving the file on the system after the test.

Using a sparse file works well on most modern filesystems:-

```c
START_TEST(test_g_file_get_size__more_than_4GiB)
{
    const char *file = "./file_5GiB.dat";
    unsigned int block_size = 4096;
    unsigned int block_count = 1310720;
    /* C90 5.2.4.2.1 guarantees long long is at least 64 bits */
    const long long expected_size =
        (long long)block_size * block_count;
    long long size;

    char cmd[256];

    /* Create a sparse file of the expected size with one block of data
     * at the end */
    g_snprintf(cmd, sizeof(cmd),
               "dd if=/dev/zero of=%s bs=%u seek=%u count=1",
               file,
               block_size,
               block_count -1);

    size = g_file_get_size(file);
    g_file_delete(file);
    ck_assert_int_eq(size, expected_size);
}
END_TEST
```

Reply:
Thanks. My file system creates a sparse file automatically when writing out consecutive zero fills.
Creating sparse files explicitly will work well.